src/parser.cr

```diff
@@ -45,12 +46,8 @@ module Mint
     # ----------------------------------------------------------------------------
 
     def raise(error : SyntaxError.class, position : Int32, raw : Hash(String, T)) forall T
-      to =
-        input[position, input.size]
-          .join
-          .split(/\s|\n|\r/)
-          .first
-          .size
```

Comment:
Imagining `raise` being raised on different places while parsing the code. This code will essentially take the rest of the chars, join them, then split them by whitespace and only take the first one. With a file of, say, 500K chars, if an error happens very early (an error for backtracking, like what happened when parsing parentheses) this would mean generating a 500K string, splitting it (generating an array with this huge string split into smaller strings)

So I changed it to scan for the next whitespace, then computing the distance. No memory allocations at all! :-)